src/lib/db.js

```diff
@@ -342,9 +342,17 @@ export default class db {
     return new Promise((resolve) => {
       database.select('slug')
       .from('teams')
-      .orderBy('name', 'asc')
+      .innerJoin('teams_authors', 'teams.id', '=', 'teams_authors.team_id')
+      .orderBy('teams_authors.team_order', 'asc')
       .then((rows) => {
-        const result = _.map(rows, (row) => {return row.slug});
+        // The SQL is returning a copy of the team for every author. This
+        // solution manually filters the results and works but it's hacky.
+        // We don't use ES6 sets to preserve the order.
+        const result = [];
+        rows.forEach((row) => {
+          if (result.indexOf(row.slug) === -1)
+            result.push(row.slug);
+        });
```

Comment:
This works, and as I said above, since there'll only be 3-5 teams, we can do whatever we want. But in theory the complexity of this approach is O(n*m) where n is number of teams and m is number of authors, where you could just make it O(m) with using an object as a hash table where you could just do `const seen = {}; if (!seen.hasOwnProperty(row.slug)) {seen[row.slug] = true;}` kinda thing